src/cdsetool/query.py

```diff
@@ -103,6 +105,19 @@ def _validate_search_terms(collection, search_terms):
         # TODO: validate patterns, minInclude, maxInclusive
 
 
+def format_dates(search_terms):
```

Comment:
This function is unnecessary. It seems to have the same concern as `_serialize_search_term`, which also converts `date` and `datetime` objects to strings